keras_cv/models/generative/stable_diffusion/stable_diffusion.py

```diff
@@ -0,0 +1,143 @@
+# Copyright 2022 The KerasCV Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import numpy as np
+import math
+
+import tensorflow as tf
+from tensorflow import keras
+
+from keras_cv.models.generative.stable_diffusion.decoder import Decoder
+from keras_cv.models.generative.stable_diffusion.diffusion_model import DiffusionModel
+from keras_cv.models.generative.stable_diffusion.text_encoder import TextEncoder
+from keras_cv.models.generative.stable_diffusion.clip_tokenizer import SimpleTokenizer
+from keras_cv.models.generative.stable_diffusion.constants import (
+    _UNCONDITIONAL_TOKENS,
+    _ALPHAS_CUMPROD,
+)
+
+MAX_PROMPT_LENGTH = 77
+
+
+class StableDiffusion:
+    def __init__(self, img_height=512, img_width=512, jit_compile=False):
+        # UNet requires multiples of 2**7 = 128
+        img_height = round(img_height / 128) * 128
+        img_width = round(img_width / 128) * 128
+        self.img_height = img_height
+        self.img_width = img_width
+        self.tokenizer = SimpleTokenizer()
+
+        # Create models
+        self.text_encoder = TextEncoder(MAX_PROMPT_LENGTH)
+        self.diffusion_model = DiffusionModel(img_height, img_width, MAX_PROMPT_LENGTH)
+        self.decoder = Decoder(img_height, img_width)
+        if jit_compile:
+            self.text_encoder.compile(jit_compile=True)
+            self.diffusion_model.compile(jit_compile=True)
+            self.decoder.compile(jit_compile=True)
+
+        # Load weights
+        text_encoder_weights_fpath = keras.utils.get_file(
+            origin="https://huggingface.co/fchollet/stable-diffusion/resolve/main/kcv_encoder.h5",
+            file_hash="4789e63e07c0e54d6a34a29b45ce81ece27060c499a709d556c7755b42bb0dc4",
+        )
```

Comment:
Should we let users load these weights from a custom directory so that they can have them stored locally if they'd like (instead of making a big network call every time a model is instantiated)

Reply:
The weights are cached automatically on local disk. We could add an option to specify the cache directory though! I'll leave that for a future update. Right now the cache is the default, at `~/.keras/datasets`.